### ISSUE TITLE
[Q] How do I apply Anime4k GLSL filters on images?

### ISSUE THREAD

**Issue by NightMachinery** [bloc97/Anime4K#238]
I want to upscale images (e.g., PNG files) using Anime4k GLSL shaders such as `https://github.com/bloc97/Anime4K/blob/master/glsl/Upscale%2BDenoise/Anime4K_Upscale_Denoise_CNN_x2_L.glsl`. Is this possible?

**Comment by Tama47** [comment]
If you got a 4K display, just drag and drop image into mpv and take screenshot.

Otherwise just use waifu2x for image.

**Comment by NightMachinery** [comment]
@Tama47 I actually want to use this for slide images extracted from lecture videos, not anime. Anime4K filters make the slides look better on mpv, and I don't really know of another option. Do you still recommend waifu2x?

I also need the process to be automatable, so manual screenshotting is not good.

**Comment by Tama47** [comment]
> Do you still recommend waifu2x?

Yes, you can batch upscale all the images at once, just be sure to use the Anime4K models as other models can take considerably longer to upscale.

**Comment by Arc-2023** [comment]
yes, you can use the new filter in ffmpeg called "libplacebo" to use glsl on video stream or just a single image.
see at :`https://stackoverflow.com/questions/76792577/how-to-use-ffmpeg-to-upscale-video-resolution-with-libplacebo/76793322#76793322`
